### PR TITLE
Add/remove Aws roles to service accounts. Bug fix

### PR DESCRIPTION
**Pull request by sarathmohan4**

### CODE REVIEW

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java [hunk 1]
```diff
@@ -1158,7 +1165,7 @@ public static boolean isSvcaccPermissionInputValid(String access) {
 				build()));
 
         if(!isSvcaccPermissionInputValid(serviceAccountGroup.getAccess())) {
-            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("{\"errors\":[\"Invalid value specified for access\"]}");
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("{\"errors\":[\"Invalid value specified for access. Valid values are read, write, deny\"]}");
```
Comment (sivanag1974) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java]: Are the valid values read,write,deny or read, reset and deny?

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java [hunk 2]
```diff
@@ -1911,4 +1920,312 @@ public ADUserAccount mapFromAttributes(Attributes attr) throws NamingException {
...
+				put(LogMessage.ACTION, "Add AWS Role to Service Account").
+				put(LogMessage.MESSAGE, "Trying to add AWS Role to Service Account").
+				put(LogMessage.APIURL, ThreadLocalContext.getCurrentMap().get(LogMessage.APIURL).toString()).
+				build()));
+
+		String roleName = serviceAccountAWSRole.getRolename();
+		String svcAccName = serviceAccountAWSRole.getSvcAccName();
+		String access = serviceAccountAWSRole.getAccess();
+		if(!isSvcaccPermissionInputValid(access)) {
+			return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("{\"errors\":[\"Invalid value specified for access. Valid values are read, write, deny\"]}");
```
Comment (sivanag1974) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java]: check the valid values, everywhere

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java [hunk 3]
```diff
@@ -1911,4 +1920,312 @@ public ADUserAccount mapFromAttributes(Attributes attr) throws NamingException {
...
+				}
+				policies.addAll(currentpolicies);
+				policies.remove(r_policy);
+				policies.remove(w_policy);
+				policies.remove(d_policy);
+				policies.add(policy);
+				policiesString = org.apache.commons.lang3.StringUtils.join(policies, ",");
+				currentpoliciesString = org.apache.commons.lang3.StringUtils.join(currentpolicies, ",");
+			} else{
+				return ResponseEntity.status(HttpStatus.UNPROCESSABLE_ENTITY).body("{\"errors\":[\"Non existing role name. Please configure it as first step\"]}");
```
Comment (sivanag1974) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java]: Can the message be improved? For example, "The AWS Role 'role_name' does not exist. Please create the role and try again!" or something better?

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/v2/controller/ServiceAccountsControllerV2.java [hunk 4]
```diff
@@ -171,4 +172,56 @@
...
 		return serviceAccountsService.getServiceAccountMeta(token, userDetails, path);
 	}
+
+	/**
+	 * Method to create an aws app role
+	 * @param token
+	 * @param awsLoginRole
+	 * @return
+	 */
+	@ApiOperation(value = "${ServiceAccountsControllerV2.createRole.value}", notes = "${ServiceAccountsControllerV2.createRole.notes}")
```
Comment (sivanag1974) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/v2/controller/ServiceAccountsControllerV2.java]: "ServiceAccountsControllerV2.createRole.value". createRole!, create What Role?

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/v2/controller/ServiceAccountsControllerV2.java [hunk 5]
```diff
@@ -171,4 +172,56 @@
...
+		return serviceAccountsService.createRole(userDetails, token, awsLoginRole);
+	}
+
+	/**
+	 * Method to create aws iam role
+	 * @param token
+	 * @param awsiamRole
+	 * @return
+	 */
+	@ApiOperation(value = "${SelfSupportController.createIamRole.value}", notes = "${SelfSupportController.createIamRole.notes}")
```
Comment (sivanag1974) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/v2/controller/ServiceAccountsControllerV2.java]: Why SelfSupportController?

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java (hunk 1 above)
Comment (sarathmohan4) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java]: Updated

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java (hunk 2 above)
Comment (sarathmohan4) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java]: Updated

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/v2/controller/ServiceAccountsControllerV2.java (hunk 4 above)
Comment (sarathmohan4) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/v2/controller/ServiceAccountsControllerV2.java]: Updated createAwsRole

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/v2/controller/ServiceAccountsControllerV2.java (hunk 5 above)
Comment (sarathmohan4) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/v2/controller/ServiceAccountsControllerV2.java]: Updated

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java (hunk 3 above)
Comment (sarathmohan4) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/ServiceAccountsService.java]: Updated

**Pull request merged**